drivers/hwinfo/hwinfo_mcux_syscon.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/hwinfo.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
```

Comment:
Would like to see the address of SYSCTL0 to get drive by devicetree, but will not hold this up.

Reply:
Thanks, I have updated the implementation to get this information from the device tree.

Reply:
I have also added the hwinfo implementation for the LPC55S6X SoC.